localstack/services/sts/sts_listener.py

```diff
@@ -26,6 +29,17 @@ def return_response(self, method, path, data, headers, response):
             if is_invalid_html_response(response.headers, response._content):
                 response.headers["Content-Type"] = "text/xml"
 
+            if headers.get("Accept") == APPLICATION_JSON:
+                # convert "Expiration" to int for JSON response format (tested against AWS)
+                # TODO: introduce a proper/generic approach that works across arbitrary date fields in JSON
```

Comment:
Is this even standardized across AWS or parseable from the schema? :thinking:

Reply:
Just chiming in here: 
Unfortunately, it's not. It's an interesting case where the AWS implementation has a specific feature which is not documented or specified anywhere (at least I couldn't find any documentation).
A symptom of that is that the boto call doesn't even work (but it also doesn't allow you to set the header easily), because it would try to parse the response as specified (which is that it has to have an XML response body).

We will have to implement this specifically for STS (or maybe we can generalize this to a very limited subset of services) in the ASF serializers once we migrate STS to ASF.

Reply:
Ok, thanks for the insights. We'll need to find a somewhat generalizable solution for a subset of services then. 👍